include/seqan3/search/configuration/mode.hpp

```diff
@@ -23,80 +25,90 @@
 namespace seqan3::detail
 {
 
-//!\brief Type for the "all" value for the configuration element "mode".
-//!\ingroup search_configuration
-struct search_mode_all {};
-//!\brief Type for the "all_best" value for the configuration element "mode".
-//!\ingroup search_configuration
-struct search_mode_all_best {};
-//!\brief Type for the "best" value for the configuration element "mode".
-//!\ingroup search_configuration
-struct search_mode_best {};
-
-} // namespace seqan3::detail
-
-namespace seqan3::search_cfg
+/*!\brief Configuration element to receive all hits within the error bounds.
+ * \ingroup search_configuration
+ */
+struct reporting_all_hits_tag : public pipeable_config_element<reporting_all_hits_tag, empty_type>
 {
+    //!\privatesection
+    //!\brief Internal id to check for consistent configuration settings.
+    static constexpr detail::search_config_id id{detail::search_config_id::mode};
```

Comment:
```suggestion
    static constexpr detail::search_config_id id{detail::search_config_id::reporting_mode};
```

Reply:
I would rename it to make it more clear that this is associated with the config element.